k8s/jiva/snapshot-cleanup.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+usage()
+{
+    echo "Usage: ./ snapshot-cleanup.sh <pv-name> <number_of_snapshots_to_delete>"
+    exit 1
+}
+
+warning()
+{
+    echo "WARNING: Snapshot cleanup involves disconnecting the application from the storage. Also, while the snapshot cleanup is in progress - you will need to ensure that the connectivity to the Kubernetes Clusters is active. In case of unexpected disconnect, you will have to run the following command to restore the volume service. snapshot-cleanup.sh <pv-name> restore_service."
+    echo "Do you want to continue (Y/N)"
+    read access
+    if [ $access == "N" ]; then
+    exit 1;
+    fi
+}
+
+# spin prints ('|','/','-','\','|') in cyclic order while snapshot deletion is in progres.
+spin()
+{
+    while true
+    do
+      stat /proc/$pid > /dev/null
+      if [ $? -ne 0 ]
+      then
+          break
+      fi
+      printf "\b${sp:i++%${#sp}:1}"
+      sleep 1
+    done
+}
+
+# delete_jiva_snapshot deletes jiva-snapshots(based on the arguments passed for number of snapshots to be deleted) using jivactl 
+delete_jiva_snapshot()
+{   
+    warning
+    min_required_snapshot=4
+    count=0
+    snapshot_number_cmd='kubectl exec -it $ctrl_pod_name -n $pvc_namespace -- jivactl snapshot ls | grep -v ID | wc -l'
+    snapshot_name_cmd='kubectl exec -it $ctrl_pod_name -n $pvc_namespace -- jivactl snapshot ls | grep -v ID | tail -1'
+
+    if [ $(eval $snapshot_number_cmd) -le $((number_of_snapshot-1)) ]; then
+        echo "Error: You have requested to delete. $number_of_snapshot. There are only $(($(eval $snapshot_number_cmd) - $min_required_snapshot)). snapshot that can be deleted snapshots available on this volume. Please re-run this command with by specifying the number of snapshots to be deleted as $(($(eval $snapshot_number_cmd) - $min_required_snapshot)) or less."
```

Comment:
"Error: You have requested to delete ${number_of_snapshot} snapshots. There are only $(($(eval $snapshot_number_cmd) - $min_required_snapshot)) snapshots on this volume. Please re-run this command by specifying the number of snapshots to be deleted as $(($(eval $snapshot_number_cmd) - $min_required_snapshot)) or less."